common/src/primitives/encoding/mod.rs

```diff
@@ -0,0 +1,67 @@
+mod errors;
+pub use errors::*;
+mod base32;
+mod bech32m;
+pub use bech32m::arbitrary_data_to_bech32m as encode;
+pub use bech32m::bech32m_to_arbitrary_data as decode;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Decoded {
+    hrp: String,
+    data: Vec<u8>,
+}
+
+impl Decoded {
+    pub fn get_hrp(&self) -> &str {
+        &self.hrp
+    }
+
+    pub fn get_data(&self) -> &[u8] {
+        &self.data
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DecodedBase32FromBech32(Decoded);
```

Comment:
A decoding process with target Base32 should have a `Vec<u5>`, not a `Vec<u8>`.

Reply:
Keep in mind that we're not required to expose this. This can be private.

Reply:
Fixed!